Correcting and adjusting Nginx docs

solidsnake1298 opened the pull request:
Correcting and adjusting Nginx docs.  Jellyfin site config should not be in conf.d.  Also making adjustments to config to address common problems.  Particularly with the COOP/COEP headers causing image searching to not show images.

solidsnake1298 commented:
Still testing to validate function vs security.

solidsnake1298 commented:
@felix920506 @thornbill

What are your opinions on the way we currently have the http vs https configs laid out?  Right now we have an http config with https parameters commented out.  But we also have a separate https example.

Should we de-clutter the http example in favor of the already separate https example?

solidsnake1298 commented:
> We should have the default be https, and tell them to comment / change certain lines if they wish to have a http only setup.

Done.  Put HTTPS prominently at the top. HTTP example is tucked away in an expandable block.

Why would the subpath examples not proxy /socket?

felix920506 commented:
why isn't the https one default for the subpath option?

solidsnake1298 commented:
> why isn't the https one default for the subpath option?

Re-arranged.

felix920506 reviewed (commented):
2 things.
1. Please help cut down on admonitions if they aren't absolutely necessary
2. Why does the subdomain config have many more security options than the subpath config?

solidsnake1298 commented:
> 1. Please help cut down on admonitions if they aren't absolutely necessary

I do agree it is excessive.  I will trim and tone down the warnings.

> 2. Why does the subdomain config have many more security options than the subpath config?

My bad.  Will make it uniform.

solidsnake1298 commented:
@felix920506 

> 1. Please help cut down on admonitions if they aren't absolutely necessary
I removed one of the cautions and changed the remaining caution to a tip and changed it a bit to tone it down.

> 2. Why does the subdomain config have many more security options than the subpath config?
Done.  Also uncommented the CSP header and added img-src to address an issue not having it causes when searching for images in the Jellyfin UI.

felix920506 commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,145 @@ title: Nginx
-
-:::
-
 :::tip
 
 The default X-Frame-Options header may cause issues with the webOS app, causing it to remain stuck at a black screen. If enabled, the default Content Security Policy may also cause issues.
 
 :::
```
A caution is more appropriate for this one but realistically it can be removed altogether IMO, there is already mentions about it in the comments in the config

felix920506 commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,145 @@ title: Nginx
     listen 80;
     listen [::]:80;
-    # server_name DOMAIN_NAME;
+    server_name jellyfin.DOMAIN.TLD;
```
Maybe using `example.com` would be better than `DOMAIN.TLD`? Also add text to tell users to replace the example domain with what they actually want

felix920506 commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,145 @@ title: Nginx
-    # listen [::]:443 ssl;
-    http2 on;
-    server_name DOMAIN_NAME;
+    server_name jellyfin.DOMAIN.TLD;
```
same as above

felix920506 commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,145 @@ title: Nginx
-    # ssl_protocols TLSv1.3 TLSv1.2;
+    ssl_protocols TLSv1.3 TLSv1.2;
+
+    ssl_certificate /etc/letsencrypt/live/DOMAIN.TLD/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/DOMAIN.TLD/privkey.pem;
+    include /etc/letsencrypt/options-ssl-nginx.conf;
+    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
+    ssl_trusted_certificate /etc/letsencrypt/live/DOMAIN.TLD/chain.pem;
```
same as above (example.com instead of DOMAIN.TLD)

felix920506 commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,145 @@ title: Nginx
     # External Javascript (such as cast_sender.js for Chromecast) must be whitelisted.
     # NOTE: The default CSP headers may cause issues with the webOS app
-    #add_header Content-Security-Policy "default-src https: data: blob: http://image.tmdb.org; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' https://www.gstatic.com https://www.youtube.com blob:; worker-src 'self' blob:; connect-src 'self'; object-src 'none'; frame-ancestors 'self'";
+    add_header Content-Security-Policy "default-src https: data: blob: ; img-src 'self' https://* ; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' https://www.gstatic.com https://www.youtube.com blob:; worker-src 'self' blob:; connect-src 'self'; object-src 'none'; frame-ancestors 'self'";
```
why are the gstatic.com and youtube.com policies needed here?

felix920506 commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,145 @@ title: Nginx
+
+### HTTP config example
+
+:::tip
+
+The following configuration is provided for ease of use only. If you are planning on exposing your server over the Internet you should setup HTTPS. [Let's Encrypt](https://letsencrypt.org/getting-started/) can provide free TLS certificates which can be installed easily via [certbot](https://certbot.eff.org/). Using only HTTP will expose passwords and API keys.
+
+:::
```
Remove admonition please

solidsnake1298 replied:
The entire thing or just the first sentence?

felix920506 commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,145 @@ title: Nginx
+    # Enforces https content and restricts JS/CSS to origin
+    # External Javascript (such as cast_sender.js for Chromecast) must be whitelisted.
+    # NOTE: The default CSP headers may cause issues with the webOS app
+    add_header Content-Security-Policy "default-src https: data: blob: ; img-src 'self' https://* ; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' https://www.gstatic.com https://www.youtube.com blob:; worker-src 'self' blob:; connect-src 'self'; object-src 'none'; frame-ancestors 'self'";
```
same as above

felix920506 commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,145 @@ title: Nginx
+server {
+    listen 80;
+    listen [::]:80;
+    server_name jellyfin.DOMAIN.TLD;
```
same as above

felix920506 reviewed (commented):
I noticed that some parts of the guide use example.org as an example domain, while others use DOMAIN.TLD, can we change them all to be uniform?

Gauvino commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,139 @@ title: Nginx
+
+:::tip
+
+If you are planning on exposing your server over the Internet you should setup HTTPS. [Let's Encrypt](https://letsencrypt.org/getting-started/) can provide free TLS certificates which can be installed easily via [certbot](https://certbot.eff.org/). Using only HTTP will expose passwords and API keys.
```
```suggestion
If you are planning on exposing your server over the Internet, you should setup HTTPS. [Let's Encrypt](https://letsencrypt.org/getting-started/) can provide free TLS certificates which can be installed easily via [certbot](https://certbot.eff.org/). Using only HTTP will expose passwords and API keys.
```

Gauvino commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,139 @@ title: Nginx
+        proxy_set_header X-Forwarded-Protocol $scheme;
+        proxy_set_header X-Forwarded-Host $http_host;
+
+        # Disable buffering when the nginx proxy gets very resource heavy upon streaming
```
```suggestion
        # Disable buffering, when the nginx proxy gets very resource heavy upon streaming
```

Gauvino commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,139 @@ title: Nginx
+
+    # use a variable to store the upstream proxy
+    # in this example we are using a hostname which is resolved via DNS
+    # (if you aren't using DNS remove the resolver line and change the variable to point to an IP address e.g `set $jellyfin 127.0.0.1`)
```
```suggestion
    # (if you aren't using DNS remove the resolver line and change the variable to point to an IP address, e.g `set $jellyfin 127.0.0.1`)
```

Gauvino commented on docs/general/networking/nginx.md:
```diff
@@ -9,83 +9,139 @@ title: Nginx
+    add_header X-XSS-Protection "0"; # Do NOT enable. This is obsolete/dangerous
+    add_header X-Content-Type-Options "nosniff";
+
+    # Permissions policy. May cause issues on some clients
```
```suggestion
    # Permissions policy. May cause issues with some clients
```

Gauvino commented on docs/general/networking/nginx.md:
```diff
@@ -115,47 +171,74 @@ server {
-    location / {
-        try_files $uri $uri/ =404;
-    }
+    # Permissions policy. May cause issues on some clients
```
```suggestion
    # Permissions policy. May cause issues with some clients
```

Gauvino commented on docs/general/networking/nginx.md:
```diff
@@ -166,66 +249,81 @@ server {
+    add_header X-XSS-Protection "0"; # Do NOT enable. This is obsolete/dangerous
+    add_header X-Content-Type-Options "nosniff";
+
+    # Permissions policy. May cause issues on some clients
```
```suggestion
    # Permissions policy. May cause issues with some clients
```

Gauvino commented on docs/general/networking/nginx.md:
```diff
@@ -166,66 +249,81 @@ server {
+    # NOTE: The default CSP headers may cause issues with the webOS app
+    add_header Content-Security-Policy "default-src https: data: blob: ; img-src 'self' https://* ; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' https://www.gstatic.com https://www.youtube.com blob:; worker-src 'self' blob:; connect-src 'self'; object-src 'none'; frame-ancestors 'self'";
+
+    # Uncomment and create directory to also host static content
```
```suggestion
    # Uncomment and create a directory to also host static content
```

Gauvino reviewed (changes requested):
Minor typo change 🫡
